pkg/dashboard/resource/v3iostream.go

```diff
@@ -113,6 +199,127 @@ func (vsr *v3ioStreamResource) getNamespaceFromRequest(request *http.Request) st
+		ContainerName: info.ContainerName,
+	}
+	getContainerContentsInput := &v3io.GetContainerContentsInput{
+		Path:             info.StreamPath,
+		GetAllAttributes: true,
+		DirectoriesOnly:  false,
+		DataPlaneInput:   dataPlaneInput,
+	}
+
+	shardLagsMap := map[string]restful.Attributes{}
+	lock := sync.Mutex{}
+
+	for {
+		response, err := v3ioContext.GetContainerContentsSync(getContainerContentsInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to get container subdirectories")
+		}
+
+		getContainerContentsOutput := response.Output.(*v3io.GetContainerContentsOutput)
+		response.Release()
+
+		errGroup, _ := errgroup.WithContextSemaphore(ctx, vsr.Logger, 10)
+		for _, content := range getContainerContentsOutput.Contents {
+			content := content
+			errGroup.Go("get-single-shard-lags", func() error {
+
+				// each file name that is just a number is a shard
+				filePath := strings.Split(content.Key, "/")
+				shardID, err := strconv.Atoi(filePath[len(filePath)-1])
+				if err != nil {
+					return nil
+				}
+
+				current, committed, err := vsr.getSingleShardLagDetails(v3ioContext,
+					info,
+					shardID,
+					dataPlaneInput)
+				if err != nil {
+					return errors.Wrapf(err, "Failed getting shard lag details, shardID: %v", shardID)
+				}
+				lock.Lock()
+				shardLagsMap[strconv.Itoa(shardID)] = restful.Attributes{
+					"lag":       current - committed,
+					"current":   current,
+					"committed": committed,
+				}
+				lock.Unlock()
+
+				return nil
+			})
+		}
+
+		if err := errGroup.Wait(); err != nil {
+			return nil, errors.Wrap(err, "Failed getting shard lags")
+		}
+
+		// if nextMarker not empty or isTruncated="true" - there are more children (need another fetch to get them)
+		if !getContainerContentsOutput.IsTruncated || len(getContainerContentsOutput.NextMarker) == 0 {
+
+			// there is no more content in the container, we can exit the loop
+			break
+		}
+		getContainerContentsInput.Marker = getContainerContentsOutput.NextMarker
+	}
+
+	return shardLagsMap, nil
+}
+
+func (vsr *v3ioStreamResource) getSingleShardLagDetails(v3ioContext v3io.Context, info v3ioStreamInfo, shardID int, dataPlaneInput v3io.DataPlaneInput) (int, int, error) {
+	response, err := v3ioContext.GetItemSync(&v3io.GetItemInput{
+		DataPlaneInput: dataPlaneInput,
+		Path:           fmt.Sprintf("%s/%d", info.StreamPath, shardID),
+		AttributeNames: []string{
+			"__last_sequence_num",
+			fmt.Sprintf("__%s_committed_sequence_number", info.ConsumerGroup),
+		},
+	})
+	if err != nil {
+		return 0, 0, err
+	}
+	defer response.Release()
+	getItemOutput := response.Output.(*v3io.GetItemOutput)
+
+	current, err := getItemOutput.Item.GetFieldInt("__last_sequence_num")
+	if err != nil {
+		return 0, 0, err
+	}
+
+	committed, err := getItemOutput.Item.GetFieldInt(fmt.Sprintf("__%s_committed_sequence_number",
+		info.ConsumerGroup))
+	if err != nil {
+		if !strings.Contains(err.Error(), "Not found") {
+			return 0, 0, err
+		}
+		committed = 0
```

Comment:
add comment that the reason it is happening is that the stream is lazly created meaning, the stream itself gets created once it has data in it. so we will have a stream, but its shards are not yet created and hence we say committed is 0